webapp/views.py

```diff
@@ -499,6 +498,32 @@ def advantage_view():
     )
 
 
+def get_machine_usage(advantage, contract):
+    """Return machine usage for the given contract as a MachineUsage object."""
+    allowances = contract.get("contractInfo", {}).get("allowances", [])
+    allowed = sum(
+        a["value"]
+        for a in allowances
+        if a["metric"] == ALLOWANCE_METRIC_ACTIVE_MACHINES
+    )
+    attached_machines = advantage.get_contract_machines(contract).get(
+        "machines"
```

Comment:
If you you provide `[]` as a default in this function, bellow you can just have `attached = len(attached_machines)`.